src/odemis/driver/xt_client.py

```diff
@@ -23,36 +22,69 @@
 from __future__ import division, print_function
 
 import logging
+import time
+from concurrent.futures import CancelledError
 
-import msgpack_numpy as mn
+import msgpack_numpy
 import zerorpc
 
-mn.patch()
+from odemis import model
+from odemis import util
+from odemis.model import CancellableThreadPoolExecutor
 
-# Necessary, otherwise nothing at all will be shown
-logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
-logging.getLogger().setLevel(logging.INFO)
+# allow to pass numpy arrays over msgpack
+msgpack_numpy.patch()
 
+XT_RUN = "run"
+XT_STOP = "stop"
+XT_CANCEL = "cancel"
 
-class MicroscopeClient:
+
+class SEM(model.HwComponent):
     """
     Class to communicate with a Microscope server via the ZeroRPC protocol.
     """
 
-    def __init__(self, server_address="tcp://192.168.1.1:4242", timeout=30):
+    def __init__(self, name, role, children, address, timeout, daemon=None,
+                 **kwargs):
         """
         Parameters
         ----------
-        server_address: str
-            server address of the Microscope server.
-        timeout: int
+        address: str
+            server address and port of the Microscope server, e.g. tcp://192.168.1.1:4242
+        timeout: float
             Time in seconds the client should wait for a response from the server.
         """
+
+        model.HwComponent.__init__(self, name, role, daemon=daemon, **kwargs)
         # set heartbeat to None on client and server side, otherwise after two missed heartbeats the client thinks the
         # connection is lost. A heartbeat happens every 5 seconds, when a function takes longer than 5 seconds to
         # respond a heartbeat is skipped. timeout controls how long a call can take to respond, the default is 30
         # seconds.
-        self.client = zerorpc.Client(server_address, heartbeat=None, timeout=timeout)
+        self.client = zerorpc.Client(heartbeat=None, timeout=timeout)
+        self.client.connect(address)
```

Comment:
Please catch exceptions, so that if the server address is not responding it raises a `HwError`. This allows to automatically display a nice message in the back-end start-up, and automatically retry.